test/io_service_tests.cpp

```diff
@@ -226,4 +226,36 @@ TEST_CASE_FIXTURE(io_service_fixture_with_threads<1>, "Many concurrent timers")
 		<< "ms");
 }
 
+TEST_CASE("io_service::process_events_until_complete(task<T>)")
+{
+	cppcoro::io_service ioService;
+
+	auto makeTask = [](cppcoro::io_service& io) -> cppcoro::task<std::string>
+	{
+		co_await io.schedule();
+		co_return "foo";
+	};
+
+	auto task = makeTask(ioService);
+
+	CHECK(ioService.process_events_until_complete(task) == "foo");
+	CHECK(ioService.process_events_until_complete(makeTask(ioService)) == "foo");
+}
+
+TEST_CASE("io_service::process_events_until_complete(shared_task<T>)")
+{
+	cppcoro::io_service ioService;
+
+	auto makeTask = [](cppcoro::io_service& io) -> cppcoro::shared_task<std::string>
+	{
+		co_await io.schedule();
+		co_return "foo";
+	};
+
+	auto task = makeTask(ioService);
+
+	CHECK(ioService.process_events_until_complete(task) == "foo");
```

Comment:
Would you be able to add an extra test in here for passing the already-completed `task` into `process_event_until_complete()` to test the code-path in the implementation where `task.is_ready()` returns true?